Packs/FetchIndicatorsFromFile/Scripts/FetchIndicatorsFromFile/FetchIndicatorsFromFile.py

```diff
@@ -0,0 +1,176 @@
+
+    if re.match(emailRegex, indicator):
+        return FeedIndicatorType.Account
+
+    else:
+        return FeedIndicatorType.Domain
+
+
+def fetch_indicators_from_file(args):
+    file = demisto.getFilePath(args.get('entry_id'))
+    file_path = file['path']
+    file_name = file['name']
+    auto_detect = True if args.get('auto_detect') == 'True' else False
+    default_type = args.get('default_type')
+    limit = args.get("limit")
+    offset = int(str(args.get("offset"))) if args.get('offset') else 0
+    skip_parsing = False
+    sheet_name = args.get('sheet_name')
+    indicator_col_num = args.get('indicator_column_number')
+    type_col_num = args.get('indicator_type_column_number')
+    starting_row = args.get('starting_row')
+
+    if file_name.endswith('xls') or file_name.endswith('csv') or file_name.endswith('xlsx'):
+
+        if not type_col_num:
+            indicator_list = xls_file_to_indicator_list(file_path, sheet_name,
+                                                        int(indicator_col_num) - 1, int(starting_row) - 1)
+
+        else:
+            indicator_list = xls_file_to_parsed_indicators(file_path, sheet_name,
+                                                           int(indicator_col_num) - 1, int(type_col_num) - 1,
+                                                           int(starting_row) - 1)
+            skip_parsing = True
+
+    else:
+        indicator_list = txt_file_to_indicator_list(file_path)
+
+    if not skip_parsing:
+
+        parsed_indicators = []
+        for indicator in indicator_list:
+            if auto_detect:
+                indicator_type = detect_type(indicator)
+
+            else:
+                indicator_type = default_type
+
+            parsed_indicators.append({
+                "Type": indicator_type,
+                "Value": indicator
+            })
+
+    else:
+        parsed_indicators = indicator_list
+
+    if limit:
+        limit = int(str(limit))
+        parsed_indicators = parsed_indicators[offset: limit + offset]
+
+    human_readable = tableToMarkdown("Indicators from {}:".format(file_path), parsed_indicators,
+                                     headers=['Value', 'Type'], removeNull=True)
+
+    if limit:
+        human_readable = human_readable + "\nTo bring the next batch of indicators run:\n!FetchIndicatorsFromFile " \
+            "limit={} offset={} entry_id={}".format(limit, int(limit) + int(offset), args.get('entry_id'))
+
+        if sheet_name:
+            human_readable = human_readable + " sheet_name={}".format(sheet_name)
+
+        if int(indicator_col_num) != 1:
+            human_readable = human_readable + " indicator_column_number={}".format(indicator_col_num)
+
+        if type_col_num:
+            human_readable = human_readable + " indicator_type_column_number={}".format(type_col_num)
+
+    return human_readable, {
+        "Indicator(val.Value == obj.Value && val.Type == obj.Type)": parsed_indicators
```

Comment:
why don't you use demisto.createindicators?

Reply:
I thought the script should behave **like** a demisto tip integration - so the context holds the indicators in that format and an associated playbook should handle the createindicators part - am I wrong? @avidan-H